metpy/plots/declarative.py

```diff
@@ -752,13 +747,49 @@ def data(self, val):
         self._data = val
         self._update_data()
 
+    @property
+    def name(self):
+        """Generate a name for the plot."""
+        if isinstance(self.field, tuple):
+            ret = ''
+            ret += ' and '.join(f for f in self.field)
+        else:
+            ret = self.field
+        if self.level is not None:
+            ret += '@{:d}'.format(self.level)
+        return ret
+
+
+@exporter.export
+class Plot2D(Plots):
+    """Represent plots of 2D scalar data."""
+
+    field = Unicode()
+    contours = Union([List(Float()), Int()], default_value=25)
```

Comment:
Do you remember why we put `contours` and `clabels` on `Plot2D` rather than on `ContourPlot` and `FilledContourPlot`?

Reply:
I did that because they were general to both, but not everything in one of them (e.g., linestyle, linewidth) would be appropriate for the other, so it would not be able to inherit one from the other.